relayer-cli/src/config.rs

```diff
@@ -4,20 +4,46 @@
 //! application's configuration file and/or command-line options
 //! for specifying it.
 
-use std::path::PathBuf;
+use std::collections::BTreeSet;
 
-use abscissa_core::{error::BoxError, EntryPoint, Options};
-
-use crate::commands::CliCmd;
+use thiserror::Error;
 
+use ibc::ics24_host::identifier::ChainId;
 pub use ibc_relayer::config::Config;
 
-/// Get the path to configuration file
-pub fn config_path() -> Result<PathBuf, BoxError> {
-    let mut args = std::env::args();
-    assert!(args.next().is_some(), "expected one argument but got zero");
-    let args = args.collect::<Vec<_>>();
-    let app = EntryPoint::<CliCmd>::parse_args_default(args.as_slice())?;
-    let config_path = app.config.ok_or("no config file specified")?;
-    Ok(config_path)
+/// Specifies all the possible syntactic errors
+/// that a Hermes configuration file could contain.
+#[derive(Error, Debug)]
+pub enum Error {
+    /// No chain was configured
+    #[error("config file does not specify any chain")]
+    ZeroChains,
+
+    /// The log level is invalid
+    #[error("config file specifies an invalid log level ('{0}'), caused by: {1}")]
+    InvalidLogLevel(String, String),
+
+    /// Duplicate chains configured
+    #[error("config file has duplicate entry for the chain with id {0}")]
+    DuplicateChains(ChainId),
+}
+
+/// Validation method for syntactic validation of the input
+/// configuration file.
+pub fn validate_config(config: &Config) -> Result<(), Error> {
+    // Check for empty or solitary chains config.
```

Comment:
what is "solitary chains"?

Reply:
That was a stale comment that I forgot to delete! Should be fixed.